bot/handlers/role_selection.go

```diff
@@ -40,46 +40,40 @@ func NewRoleSelectionHandler(guildID string, config *config.RoleSelectionConfig)
 }
 
 // ReconcileConfig reconciles the role selection configuration with the Discord guild.
-// TODO: This function is not fully implemented and needs to be more reliable.
 func (r *RoleSelectionHandler) ReconcileConfig(s *discordgo.Session) error {
-	r.reconciling.RLock()
-	chList, err := s.GuildChannels(r.guildID)
+	r.reconciling.Lock()
+	defer r.reconciling.Unlock()
+
+	roleChannel, err := s.Channel(r.config.Channel.ID)
 	if err != nil {
-		r.logger.Error("error getting channels for guild", zap.String("guildID", r.guildID), zap.Error(err))
-		r.reconciling.RUnlock()
+		r.logger.Error("error getting channel", zap.String("channelID", r.config.Channel.ID), zap.Error(err))
 		return err
 	}
-	roleChExists := false
-	var roleChannel *discordgo.Channel
-	for _, ch := range chList {
-		if ch.ID == r.config.Channel.ID {
-			r.logger.Debug("role channel exists!", zap.String("channel", ch.Name))
-			roleChExists = true
-			roleChannel = ch
-		}
+
+	messages, err := s.ChannelMessages(roleChannel.ID, 100, "", "", "")
+	if err != nil {
+		r.logger.Error("error getting messages for channel", zap.String("channelID", roleChannel.ID), zap.Error(err))
+		return err
 	}
-	r.reconciling.RUnlock()
-	if !roleChExists {
-		r.logger.Debug("channel does not yet exist. Wat?!")
-		// create channel
-	}
-	messagesCreated := 0
-	// TODO: Do this more reliably instead of relying on counts
-	if roleChannel.MessageCount == 0 && len(r.prompts) < len(r.config.Prompts) {
-		r.logger.Debug("locking for message creation")
-		r.reconciling.Lock()
-		defer func() {
-			r.logger.Debug("unlocking msg creation")
-			r.reconciling.Unlock()
-		}()
-		for _, p := range r.config.Prompts {
+
+	for _, p := range r.config.Prompts {
```

Comment:
I sense a future bug with duplicate messages. But that might be existing anyway 😅